### PR TITLE
latest API changes

### PR DESCRIPTION
**Pull request by magnurud**
- add data bundle endpoints
- Finish all endpoints for data bundles
- add datasetId as optional parameter to UpdateDocument
- add deleteAll for documents
- fixes after testing on real API
- temporary fixes
- use deleteAll flag when deleting documents from DeleteDataset and DeleteBatch
- add delete documents
- update version

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v1.7.2
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+      - name: Setup Prism
+        run: make prism-start
+      - name: Run Tests
+        run: make test
```
Comment (augustak) [inline .github/workflows/run_tests.yml:22]:
```suggestion
        run: make test
        
```

File: Lucidtech/Las/Client.cs [hunk 2]
```diff
@@ -458,6 +461,34 @@ public object UpdateDocument(string documentId, List<Dictionary<string, string>>
             }
 
             RestRequest request = ClientRestRequest(Method.DELETE, "/documents", null, queryParams);
+            var objectResponse = ExecuteRequestResilient(RestSharpClient, request);
+            var response = JsonSerialPublisher.ObjectToDict<Dictionary<string, object>>(objectResponse);
+
+            var documentsDeleted = new List<object>();
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:467]: Can this be defined inside the if scope?

File: Lucidtech/Las/Client.cs [hunk 3]
```diff
@@ -525,18 +556,11 @@ public object CreateBatch(string? name = null, string? description = null)
         /// <param name="deleteDocuments">Set to true to delete documents in batch before deleting batch</param>
         /// <returns>Batch response from REST API</returns>
         public object DeleteBatch(string batchId, bool deleteDocuments = false) {
+
             if (deleteDocuments == true) {
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:560]: Do we need == true?

File: Lucidtech/Las/Client.cs [hunk 4]
```diff
@@ -635,18 +659,11 @@ public object CreateDataset(string? name = null, string? description = null)
         /// <param name="deleteDocuments">Set to true to delete documents in dataset before deleting dataset</param>
         /// <returns>Dataset response from REST API</returns>
         public object DeleteDataset(string datasetId, bool deleteDocuments = false) {
+
             if (deleteDocuments == true) {
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:663]: Do we need == true?

File: Lucidtech/Las/Client.cs [hunk 5]
```diff
@@ -916,6 +933,90 @@ public object CreateDataset(string? name = null, string? description = null)
...
+        /// <param name="modelId">Id of the model</param>
+        /// <param name="maxResults">Number of items to show on a single page</param>
+        /// <param name="nextToken">Token to retrieve the next page</param>
+        /// <returns>
+        /// JSON object with two keys:
+        /// - "dataBundles" which contains a list of data bundle objects
+        /// - "nextToken" allowing for retrieving the next portion of data
+        /// </returns>
+        public object ListDataBundles(string modelId, int? maxResults = null, string? nextToken = null) {
+            var queryParams = new Dictionary<string, object?>();
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:976]: Why is object optional here, does that make sense?

**Review by augustak — CHANGES_REQUESTED** [review]

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (magnurud) [reply .github/workflows/run_tests.yml:22]: an extra empty line?

File: Lucidtech/Las/Client.cs (hunk 5 above)
Comment (magnurud) [reply Lucidtech/Las/Client.cs:976]: Because it can be empty if no argument is provided

**Review by augustak — APPROVED** [review]

**Pull request merged**